tap-setup-scripts/src/resources/tap-gui-viewer-secret.yaml

```diff
@@ -0,0 +1,9 @@
+---
+apiVersion: v1
+kind: Secret
```

Comment:
*question*: Instead of creating that secret here, could we add this as an overlay to the `tap-gui` package?

With that, we wouldn't have to do those "one-off" `kubectl` calls to create and delete the secret; the kapp-controller should handle all that for us, and we'd don't have to "reimplement" all that in our cloud init script.

Most/all(?) parts for that should already be there and what I am thinking of is something like:
1. add another overlay secret to [package_overlay_secrets](https://github.com/vmware-tap-on-public-cloud/quickstart-vmware-tanzu-application-platform/blob/d148889f3637bced3180a3a31065e7d2248edd3f/tap-setup-scripts/src/resources/package_overlay_secrets.yaml), similar to the the [telemetry overlays](https://github.com/vmware-tap-on-public-cloud/quickstart-vmware-tanzu-application-platform/blob/d148889f3637bced3180a3a31065e7d2248edd3f/tap-setup-scripts/src/resources/package_overlay_secrets.yaml#L24-L32)
2. In [package_overlay_apply](https://github.com/vmware-tap-on-public-cloud/quickstart-vmware-tanzu-application-platform/blob/d148889f3637bced3180a3a31065e7d2248edd3f/tap-setup-scripts/src/resources/package_overlay_apply.yaml) add that overlay secret to the `pkgi`, similar to the [telemetry overlays](https://github.com/vmware-tap-on-public-cloud/quickstart-vmware-tanzu-application-platform/blob/d148889f3637bced3180a3a31065e7d2248edd3f/tap-setup-scripts/src/resources/package_overlay_apply.yaml#L8-L12)

Reply:
yes, there are multiple ways to do the same thing. An overlay is one of them.  I guess we can do that in another exercise in a different PR.

Reply:
*non-blocking*

To add more context why I believe overlays should generally be preferred:

- any change we do, any objects we modify, any objects we add, are now part of the `pkgi` and its `app`. Thus the kapp-controller handles their lifecycle, e.g. cleanup.
- with overlays we can describe the desired state from the get go, thus we don't have to deploy something with a "not desired" state, to later change things up, redeploy, and hope it can transition from that intermediate state to the actual desired state.
- with the use of kapp's changegroups/changerules we can express ordering of upserts & deletes of objects. With kapp's waitrules we can have influence on when we deem objects to have reconciled correctly. We can "just" instruct the kapp-controller to do these kinds of things for us, we don't have to implement them with busy loops.

While the later points are not relevant for this specific case, the first one is.